contrib/native/client/src/clientlib/drillClientImpl.cpp

```diff
@@ -179,9 +179,11 @@ connectionStatus_t DrillClientImpl::sendHeartbeat(){
 }
 
 void DrillClientImpl::resetHeartbeatTimer(){
-    m_heartbeatTimer.cancel();
-    DRILL_MT_LOG(DRILL_LOG(LOG_TRACE) << "Reset Heartbeat timer." << std::endl;)
-    startHeartbeatTimer();
+    if (DrillClientConfig::getHeartbeatFrequency() > 0) {
+        m_heartbeatTimer.cancel();
```

Comment:
You are right, cancelling is not required because `timer.expires_from_now(...)` cancels the timer as well. So I will remove this method and move the check to `startHeartbeatTimer`.